core_contracts/rewards/rewards.py

```diff
@@ -69,10 +69,16 @@ def on_install(self, _governance: Address) -> None:
 
     def on_update(self) -> None:
         super().on_update()
-        total_dist = self._data_source_db['sICX/bnUSD'].total_dist[1]
-        total_value = self._data_source_db['sICX/bnUSD'].total_value[1]
-        self.Report(1, total_dist, total_value)
-        self._data_source_db['sICX/bnUSD'].day.set(2)
+        self.Report(self._get_day(), 'test', self._total_dist.get(), 0)
+
+    @external
+    @only_governance
+    def setDay(self, day: int) -> None:
+        for name in self._data_source_db:
+            total_dist = self._data_source_db[name].total_dist[day - 1]
+            total_value = self._data_source_db[name].total_value[day - 1]
+            self.Report(day - 1, name, total_dist, total_value)
+            self._data_source_db[name].day.set(day)
```

Comment:
Shouldn't you use the `set_day` method from `DataSource` instead ?

Reply:
Done. Thanks.